proposals/lip-0038.md

```diff
@@ -79,41 +80,46 @@ The Lisk protocol could contain several transaction types that perform such a re
 
 We only consider signature aggregation for the case where several signatures for the same message are aggregated.
 
-Each aggregate signature needs to be accompanied by some information that specifies the set of public keys that correspond to the aggregate signature. Here, this is realized using a bitmap. Assume that `keyList` is a list that includes all potential public keys that could participate in the signature aggregation. The entries must be pairwise distinct. Moreover, let `pubKeySignaturePairs` be a list of pairs of public keys and signatures where all signatures belong to the same message, and all public keys are unique and contained in `keyList`. Then, the corresponding aggregate signature and bitmap can be computed via `createAggSig(keysList, pubKeySignaturePairs)` as in the pseudo code below. To verify if a signature is an aggregate signature of a binary message `message`, the function `verifyAggSig` can be used. `verifyAggSig(keysList, aggregationBits, signature, tag, networkIdentifier, message)` returns `VALID` if and only if `signature` is an aggregate signature of the message `message` for the message tag `tag`, the network identifier `networkIdentifier` and for the public keys in `keyList` defined by `aggregationBits`.
+Each aggregate signature needs to be accompanied by some information that specifies the set of public keys that correspond to the aggregate signature. Here, this is realized using a bitmap. Assume that `keyList` is a list that includes all potential public keys that could participate in the signature aggregation. The entries must be pairwise distinct. Moreover, let `pubKeySignaturePairs` be a list of pairs of public keys and signatures where all signatures belong to the same message, and all public keys are unique and contained in `keyList`. Then, the corresponding aggregate signature and bitmap can be computed via `createAggSig(keysList, pubKeySignaturePairs)` as in the pseudo code below. To verify if a signature is an aggregate signature of a binary message `message`, the function `verifyAggSig` can be used. `verifyAggSig(keysList, aggregationBits, signature, tag, chainID, message)` returns `True` if and only if `signature` is an aggregate signature of the message `message` for the message tag `tag`, the chain identifier `chainID` and for the public keys in `keyList` defined by `aggregationBits`.
 
 ```python
-createAggSig(keysList, pubKeySignaturePairs):
+def createAggSig(keysList: list[bytes], pubKeySignaturePairs: list[tuple[bytes, bytes]]) -> tuple[bytes, bytes]:
     aggregationBits = byte string of length ceil(length(keyList)/8) with all bytes set to 0
```

Comment:
integer division `//8` ?

Reply:
but here we take the ceil rather than the floor. For sure there is a nice way of doing it, I was just trying not to overthink it

Reply:
Ah, true, all good.